Android/app/src/main/java/app/intra/MainActivity.java

```diff
@@ -134,13 +134,27 @@ private void openUrl(String url) {
     startActivity(i);
   }
 
+  @Override
+  public void onSharedPreferenceChanged(SharedPreferences preferences, String key) {
+    if (PersistentState.URL_KEY.equals(key)) {
+      updateServerName();
+    }
+  }
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
+    // Sync old settings into new preferences if necessary.
+    PersistentState.syncLegacyState(this);
+
     // Export defaults into preferences.  See https://developer.android.com/guide/topics/ui/settings#Defaults
     PreferenceManager.setDefaultValues(this, R.xml.preferences, false);
 
+    // Registers this class as a listener for user preference changes.
+    PreferenceManager.getDefaultSharedPreferences(this).
```

Comment:
Is there a risk of a memory leak here since you don't unregister? Is that automatic? I would guess it's OK if the non-service activity gets closed, but might be worth checking if you haven't already?

Reply:
OK, I added an unregister call.